xapian-core/tests/api_sorting.cc

```diff
@@ -319,3 +319,51 @@ DEFINE_TESTCASE(sortfunctorremote1, remote) {
     );
     return true;
 }
+
+DEFINE_TESTCASE(replace_weights1, backend) {
+    Xapian::Database mydb(get_database("apitest_onedoc"));
+    Xapian::Enquire enquire(mydb);
+    enquire.set_query(Xapian::Query("word"));
+    Xapian::MSet mymset = enquire.get_mset(0, 10);
+    // old_max_possible, max_attained = 0.269763689697702
+    double old_max_possible = mymset.get_max_possible();
+    double new_weight = 0.125;
+    double weights[1] = {new_weight};
```

Comment:
If you make an array `static` and `const` then it can be initialised at compile time rather than at run time.  In this case you'll also want to make `new_weight` `const`.

Also, it's probably better to let the compiler fill in the number of elements, so:

```
static const weights[] = {new_weight};
```

More instances in the testcases below.